src/lib/sidenav/sidenav.ts

```diff
@@ -110,7 +112,22 @@ export class MdSidenav implements AfterContentInit {
    * @param _elementRef The DOM element reference. Used for transition and width calculation.
    *     If not available we do not hook on transitions.
    */
-  constructor(private _elementRef: ElementRef) {}
+  constructor(private _elementRef: ElementRef) {
+    this.onOpen.subscribe(() => {
+      this._elementFocusedBeforeSidenavWasOpened = document.activeElement as HTMLElement;
```

Comment:
for things such as 
```
this._elementRef.nativeElement.focus();
```
you can use renderer 
```
this.renderer.invokeElementMehod(this._elementRef.nativeElement, 'focus');
``` 
Core already have different renderer implementations for browser and server envs.

But `Renderer` doesn't have methods to retrieve dom elements and afaik the recommended approach to use DomAdapter.